core/src/test/scala/org/apache/spark/DebugFilesystem.scala

```diff
@@ -31,21 +30,29 @@ import org.apache.spark.internal.Logging
 
 object DebugFilesystem extends Logging {
   // Stores the set of active streams and their creation sites.
-  private val openStreams = new ConcurrentHashMap[FSDataInputStream, Throwable]()
+  private val openStreams = mutable.Map.empty[FSDataInputStream, Throwable]
 
-  def clearOpenStreams(): Unit = {
+  def addOpenStream(stream: FSDataInputStream): Unit = synchronized {
```

Comment:
It's a little safer and tidier to synchronize on `openStream` rather than the containing `object`. Looks good to me though.